### PR TITLE
Add page for DateTimePicker component

### PR DESCRIPTION
**Pull request by chiaramooney**
datetimepicker-RNgallery

### CODE REVIEW

File: windows/rngallery/AutolinkedNativeModules.g.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // AutolinkedNativeModules.g.cpp contents generated by "react-native autolink-windows"
-
+// clang-format off
```
Comment (chrisglein) [inline windows/rngallery/AutolinkedNativeModules.g.cpp:2]: Why was this necessary?

File: src/examples/DateTimePickerExamplePage.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        value={date}
+        mode="date"
+        display="default"
+        maximumDate={new Date(2050, 12, 31)}
+        minimumDate={new Date(1950, 1, 1)}
+        dayOfWeekFormat={'{dayofweek.abbreviated(3)}'}
+      />
+      <Text>Sample Time Picker</Text>
+      <DateTimePicker
+        mode="time"
```
Comment (asklar) [inline src/examples/DateTimePickerExamplePage.tsx:31]: consider adding another one for date too

File: src/examples/DateTimePickerExamplePage.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        value={time}
+        style={{width: 300, opacity: 1, height: 30}}
+        is24Hour={false}
```
Comment (asklar) [inline src/examples/DateTimePickerExamplePage.tsx:34]: consider binding this to a toggleswitch's state?

**Review by asklar — APPROVED** [review]

File: src/examples/DateTimePickerExamplePage.tsx (hunk 2 above)
Comment (chrisglein) [reply src/examples/DateTimePickerExamplePage.tsx:31]: Isn't there a date one right above this?

Comment (asklar) [reply src/examples/DateTimePickerExamplePage.tsx:31]: reading is hard

File: src/examples/DateTimePickerExamplePage.tsx (hunk 3 above)
Comment (chrisglein) [reply src/examples/DateTimePickerExamplePage.tsx:34]: Good question... how much interactivity should there be on each page. If the reference is XAML Controls Gallery these aren't meant to be comprehensive test pages like we might have for the internal tests. Looking at the XAML Controls Gallery page for DatePicker and TimePicker there's no options to toggle the UI settings, but there are multiple instances of the control (with markup) to showcase its options.

image

image

**Comment by rectified95** [comment]
I think you've added some .binlog files to the PR by accident

File: src/examples/DateTimePickerExamplePage.tsx (hunk 3 above)
Comment (rectified95) [reply src/examples/DateTimePickerExamplePage.tsx:34]: Let's think how many options we want to display. On the one hand it makes sense to make it rich, but then there're already the example app in the community module repos (maybe the way is to copy those samples here? they should work OOTB).

File: windows/rngallery/AutolinkedNativeModules.g.cpp (hunk 1 above)
Comment (chiaramooney) [reply windows/rngallery/AutolinkedNativeModules.g.cpp:2]: Not sure why that was added to the generated file. I'll remove it now.

File: windows/rngallery/App.cpp [hunk 4]
```diff
@@ -41,6 +41,7 @@ App::App() noexcept
 
     PackageProviders().Append(make<ReactPackageProvider>()); // Includes all modules in this project
     PackageProviders().Append(winrt::CheckboxWindows::ReactPackageProvider());
+    PackageProviders().Append(winrt::DateTimePicker::ReactPackageProvider());
```
Comment (kmelmon) [inline windows/rngallery/App.cpp:44]: Please add a TODO to remove these lines of code once issue [https://github.com/microsoft/react-native-gallery/issues/9](https://github.com/microsoft/react-native-gallery/issues/9) is fixed.

**Review by kmelmon — APPROVED** [review]
:shipit:

**Comment by kmelmon** [comment]
@chiaramooney Is anything blocking this from making forward progress?  I'd love to see this get checked in soon.

**Comment by chiaramooney** [comment]
Latest Progress:

- Separated out DatePickerExamplePage and TimePickerExamplePage into two separate pages to more closely match the setup of the XAML Controls Gallery. 
- Crafted first version of styling/formatting of the Component Pages, worked to closely match style and structure of XAML Controls Gallery.
- Added feature of displayed JSX test being copied to clipboard when clicked. 
- Add TemplateExamplePage as development option for those who are adding a page, simply make a copy of the TemplateExamplePage contents and follow its directions to fill the page with the correct information for your component. 

**_RN Gallery TimePicker (left) vs. XAML Controls Gallery TimePicker (right)_**
TimePicker-SideBySide
**_RN Gallery DatePicker (left) vs. XAML Controls Gallery DatePicker (right)_**
DatePicker-SideBySide
**_RN Gallery Template_**
Template

**_TODO:_** 
- Investigate if there is currently windows support for syntax highlighting/markdown for JSX text on component pages. 
- Missing: XAML Controls Gallery displays JSX source code in one line and if the window is too narrow, the user can scroll horizontally.
- Investigate if there is currently windows support for hyperlinks in Text component. 
- Add scrolling to pages.
- Refine action of copying snippet of JSX to clipboard, should a clickable icon be added, or a popup message saying copy was successful, etc?

File: src/RNGalleryList.ts [hunk 5]
```diff
@@ -13,10 +16,22 @@ export const RNGalleryList: Array<IRNGalleryExample> = [
     key: 'Welcome',
     component: WelcomePage,
   },
+  {
+    key: 'Template',
+    component: TemplateExamplePage,
+  },
```
Comment (kmelmon) [inline src/RNGalleryList.ts:22]: Thank you for adding the template!

File: src/examples/TemplateExamplePage.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    };
+
+    const copyExample2ToClipboard = () => {
+      Clipboard.setString(example2jsx);
+    };
+    
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>Template</Text>
+      <Text style={styles.description}>Add component description here. See XAML Controls Gallery for description contents if available.</Text>
```
Comment (kmelmon) [inline src/examples/TemplateExamplePage.tsx:26]: >XAML Controls Gallery [](start = 75, length = 21)

Can this be a hyperlink using the Linking module?  RNW does appear to support it.  See Linking module docs:  [https://reactnative.dev/docs/linking](https://reactnative.dev/docs/linking)

Comment (chiaramooney) [reply src/examples/TemplateExamplePage.tsx:26]: Yea I'll have to investigate it we can get hyperlinks at this time. I recall Jon saying something before about that not working. I'll look into it next.

**Comment by chiaramooney** [comment]
DateTimePicker missing typings. PR out now to fix module [here](https://github.com/react-native-datetimepicker/datetimepicker/pull/346).

**Comment by kmelmon** [comment]
@chiaramooney thanks for updating the typings on the module!  Feel free to temporarily add @ts-ignore statements to get past the typescript errors.  See https://github.com/microsoft/react-native-gallery/blob/00694a1f8bd9c4568d484ff045743b81a49ee2b0/src/App.tsx#L28 as an example.

**Comment by chiaramooney** [comment]
PR for missing typings has been approved. New release should be coming out this week.

**Comment by acoates-ms** [comment]
@chiaramooney - I have a change I'd like to build on top of this change.  If you can still make the original checkin with the @ts-ignore's, I'd appreciate it.

**Comment by chiaramooney** [comment]
> @chiaramooney - I have a change I'd like to build on top of this change. If you can still make the original checkin with the @ts-ignore's, I'd appreciate it.

@ts-ignore change added now!
